src/frontend/planner_test/tests/testdata/watermark.yaml

```diff
@@ -76,6 +76,25 @@
       | └─StreamTableScan { table: t1, columns: [t1.ts, t1.v1, t1.v2, t1._row_id], pk: [t1._row_id], dist: UpstreamHashShard(t1._row_id) }
       └─StreamExchange { dist: HashShard(t2.ts) }
         └─StreamTableScan { table: t2, columns: [t2.ts, t2._row_id], pk: [t2._row_id], dist: UpstreamHashShard(t2._row_id) }
+- name: band hash join
+  sql: |
+    create table t1 (ts timestamp with time zone, v1 int, v2 int, watermark for ts as ts - INTERVAL '1' SECOND) append only;
+    create table t2 (ts timestamp with time zone, v1 int, v2 int, watermark for ts as ts - INTERVAL '1' SECOND) append only;
+    select t1.ts as t1_ts, t1.v1 as t1_v1, t1.v2 as t1_v2, t2.ts as t2_ts, t2.v1 as t2_v1, t2.v2 as t2_v2 from t1 left outer join t2 on (t1.v1 = t2.v1 and (t1.ts >= t2.ts + INTERVAL '1' SECOND) and (t2.ts >= t1.ts + INTERVAL '1' SECOND));
+  logical_plan: |
+    LogicalProject { exprs: [t1.ts, t1.v1, t1.v2, t2.ts, t2.v1, t2.v2] }
+    └─LogicalJoin { type: LeftOuter, on: (t1.v1 = t2.v1) AND (t1.ts >= (t2.ts + '00:00:01':Interval)) AND (t2.ts >= (t1.ts + '00:00:01':Interval)), output: all }
+      ├─LogicalScan { table: t1, columns: [t1.ts, t1.v1, t1.v2, t1._row_id] }
+      └─LogicalScan { table: t2, columns: [t2.ts, t2.v1, t2.v2, t2._row_id] }
+  stream_plan: |
+    StreamMaterialize { columns: [t1_ts, t1_v1, t1_v2, t2_ts, t2_v1, t2_v2, t1._row_id(hidden), t2._row_id(hidden)], pk_columns: [t1._row_id, t2._row_id, t1_v1], pk_conflict: "no check", watermark_columns: [t1_ts, t2_ts] }
+    └─StreamHashJoin { type: LeftOuter, predicate: t1.v1 = t2.v1 AND (t1.ts >= $expr2) AND ($expr1 <= t2.ts), output_watermarks: [t1.ts, t2.ts], output: [t1.ts, t1.v1, t1.v2, t2.ts, t2.v1, t2.v2, t1._row_id, t2._row_id] }
```

Comment:
Could you please rename the `StreamHashJoin` to `StreamIntervalJoin` if it can be converted to an interval join?

Reply:
> Could you please rename the `StreamHashJoin` to `StreamIntervalJoin` if it can be converted to an interval join?

What is the definition of a(n) interval/band join? A hash join which has no watermark on equal condition but can generate watermark on both sides by inequality?

Reply:
> > What is the definition of a(n) interval/band join? A hash join which has no watermark on equal condition but can generate watermark on both sides by inequality?
> 
> An interval join should contain two range conditions, which are watermark columns. Our primary purpose of the interval join right now is used to clean the state. If a band join contains two range conditions, which aren't watermark columns, there is no performance improvement right now, so I think we can just use regular hash join instead.

ok, fixed.

Reply:
So the difference between the two names is whether watermarks on the range condition column can do state cleaning?

Reply:
> So the difference between the two names is whether watermarks on the range condition column can do state cleaning?

When we talk about watermarks and state cleaning, I think using `interval join` is probably better, because Flink has already used it and users might be more familiar with `interval join`. In cases where range conditions lack watermarks, my suggestion is to refer to it as a `band join`. However, since our current implementation cannot utilize range conditions without watermarks, there are no discernible distinctions between it and the standard hash join. Therefore, let us continue referring to it as a `hash join`.

Reply:
My primary concern is that currently the name is 1 - 1 mapping to an executor. If we want to show users the executor is doing state cleaning using band condition, we may add a filed in the description instead of change the name.

Reply:
> My primary concern is that currently the name is 1 - 1 mapping to an executor. If we want to show users the executor is doing state cleaning using band condition, we may add a filed in the description instead of change the name.

+1 for adding more fields to tell which conditions are used to state cleaning.